### PR TITLE
Add --minimal option for check-ugrade command

### PR DESCRIPTION
**Pull request by j-mracek**
It is required for a feature parity with upgrade command where the option is already available.

Related: https://issues.redhat.com/browse/RHEL-38804

CI: https://github.com/rpm-software-management/ci-dnf-stack/pull/1511

### CODE REVIEW

File: dnf5/commands/check-upgrade/check-upgrade.cpp [hunk 1]
```diff
@@ -153,8 +176,13 @@ void CheckUpgradeCommand::run() {
             std::move(advisories.value()), installed_query, libdnf5::sack::QueryCmp::GTE);
     }
 
-    // Last, only include latest upgrades
-    upgrades_query.filter_latest_evr();
+    if (minimal) {
```
Comment (kontura) [inline dnf5/commands/check-upgrade/check-upgrade.cpp:179]: I believe this should be `minimal->get_value()`.
It is also breaking a couple of tests.

File: dnf5/commands/check-upgrade/check-upgrade.cpp [hunk 2]
```diff
@@ -46,6 +46,17 @@ void CheckUpgradeCommand::set_argument_parser() {
     auto & cmd = *get_argument_parser_command();
     cmd.set_description("Check for available package upgrades");
 
+    minimal = dynamic_cast<libdnf5::OptionBool *>(
+        parser.add_init_value(std::unique_ptr<libdnf5::OptionBool>(new libdnf5::OptionBool(false))));
+    auto minimal_opt = parser.add_new_named_arg("minimal");
+    minimal_opt->set_long_name("minimal");
+    minimal_opt->set_description(
+        "Reports the lowest versions of packages that fix a bugfix, enhancement or a fix for a security issue "
+        "(security) on the system.");
```
Comment (kontura) [inline dnf5/commands/check-upgrade/check-upgrade.cpp:55]: Both this description and the man page don't mention the `newpackage` advisory type which is also used.

What about to change it a bit to: `Reports the lowest versions of packages that fix advisories of type bugfix, enhancement, security, or newpackage.`

Anyway I do like your description much more than the one in `upgrade` command which is quite ambiguous. Could they be unified? [There is even a TODO for it.](https://github.com/rpm-software-management/dnf5/blob/main/dnf5/commands/upgrade/upgrade.cpp#L47)

**Review by kontura — CHANGES_REQUESTED** [review]

File: dnf5/commands/check-upgrade/check-upgrade.cpp (hunk 1 above)
Comment (j-mracek) [reply dnf5/commands/check-upgrade/check-upgrade.cpp:179]: May I ask you to explain your comment in more details?

Comment (j-mracek) [reply dnf5/commands/check-upgrade/check-upgrade.cpp:179]: There is an update for tests - https://github.com/rpm-software-management/ci-dnf-stack/pull/1511

Comment (j-mracek) [reply dnf5/commands/check-upgrade/check-upgrade.cpp]: I've got it, the test was really incorrect. The test was for pointer and not for bool value in the pointer of BoolOption. Fixed

**Comment by kontura** [comment]
Great, thank you.

**Pull request merged**